src/App.tsx

```diff
@@ -1,319 +1,168 @@
+import { Header } from './features/ui/header';
+import {
+  SwapContractAddressStore,
+  useAppStore,
+  useChainStore,
+  useSwapContractAddressStore,
+} from './store/store';
+import { useDomainInfo } from './hooks/useDomainInfo';
+import { useFormatSchemaValidationErrors } from './hooks/useFormatSchemaValidationErrors';
+import { LoadingOrFailed } from './features/ui/loadingOrFailed';
+import { ErrorDisplay } from './features/ui/errorDisplay';
+import { NoErrorDisplay } from './features/ui/noErrorDisplay';
+import { JsonDataDisplay } from './features/ui/jsonDataDisplay';
 
 function App() {
-  const [inputType, setInputType] = useState<InputType>(InputType.JSON);
-  const [jsonString, setJsonString] = useState<undefined | string>(undefined);
-  const [urlString, setUrlString] = useState<string | undefined>(undefined);
-  const [parsedJson, setParsedJson] = useState<
-    undefined | Partial<ParsedJsonParams>
-  >(undefined);
-  const [decompressedJson, setDecompressedJson] = useState<string | undefined>(
-    undefined
-  );
-  const [swapContractAddress, setSwapContractAddress] = useState<
-    string | undefined
-  >(undefined);
-  const [selectedChainId, setSelectedChainId] = useState<number | undefined>();
-  const [errors, setErrors] = useState<string[]>([]);
-  const [isEnableCheck, setIsEnableCheck] = useState(false);
-  const [isNoErrors, setIsNoErrors] = useState(false);
-
-  const decompressedOrderFromUrl = useDecompressedOrderFromUrl(urlString);
-
-  const chainId = parsedJson?.chainId
-    ? Number(parsedJson?.chainId)
-    : selectedChainId;
-
-  const {
-    senderWallet,
-    nonce,
-    expiry,
-    signerWallet,
-    signerToken,
-    signerAmount,
-    senderToken,
-    senderAmount,
-    v,
-    r,
-    s,
-  } = useJsonValues({ inputType, parsedJson, decompressedOrderFromUrl });
-
-  const checkFunctionArgs: CheckFunctionArgs = [
-    (senderWallet as `0x${string}`) || zeroAddress,
-    nonce || BigInt(0),
-    expiry || BigInt(0),
-    (signerWallet as `0x${string}`) || zeroAddress,
-    (signerToken as `0x${string}`) || zeroAddress,
-    signerAmount || BigInt(0),
-    (senderToken as `0x${string}`) || zeroAddress,
-    senderAmount || BigInt(0),
-    v || 0,
-    r || '0x',
-    s || '0x',
-  ];
+  const [urlMode, setUrlMode] = useState<boolean>(false);
+  const [orderText, setOrderText] = useState<string>('');
 
-  const {
-    data: checkFunctionData,
-    isLoading: isLoadingCheck,
-    error: errorCheck,
-  } = useContractRead({
-    chainId,
-    abi,
-    address: swapContractAddress as `0x${string}`,
-    functionName: 'check',
-    args: checkFunctionArgs,
-    enabled: isEnableCheck,
-  });
-
-  const { data: protocolFee, isLoading: isLoadingProtocolFee } =
-    useContractRead({
-      chainId,
-      abi,
-      address: swapContractAddress as `0x${string}`,
-      functionName: 'protocolFee',
-    });
+  const { isCheckEnabled, setIsCheckEnabled } = useAppStore();
```

Comment:
Will remove it. The reason I had this state was because `useValidateOrder` was running whenever there was an `onChange` event in the textarea. This `isCheckEnabled` state was used to hide the order details from the UI until the submit button was clicked. I'm sure there was a better way to implement that, but without a submit button, this is entirely useless, so I'll remove it